modules/datastore/tests/src/Unit/Storage/DatabaseTableTest.php

```diff
@@ -414,6 +427,27 @@ public function testQueryColumnNotFound() {
     $databaseTable->query($query);
   }
 
+  /**
+   *
+   */
+  public function testNoFulltextIndexFound() {
+    $query = new Query();
+
+    $connectionChain = $this->getConnectionChain()
+      ->add(Connection::class, 'select', Select::class, 'select_1')
+      ->add(Select::class, 'fields', Select::class)
+      ->add(Select::class, 'condition', Select::class)
+      ->add(Select::class, 'execute', new DatabaseExceptionWrapper("SQLSTATE[HY000]: General error: 1191 Can't find FULLTEXT index matching the column list..."));
```

Comment:
I see your point... FWIW the only part being matched is "Can't find FULLTEXT index matching the column list" which I think is unlikely to change anytime soon. And if it did it would just fall back to our generic error message which wouldn't be a big deal. 

I'm hesitant because we don't actually have any functional tests for the datastore right now and there's a fairly high LOE for adding that into this PR, and I don't know that this particular thing justifies doing that. We should reexamine testing for the datastore API but IMO that's out of scope for this ticket. We can discuss further if you feel strongly though.

Reply:
No, I kind of agree - we should hold off for now, at least pending datastore refactor efforts